redisson/src/main/java/org/redisson/config/Config.java

```diff
@@ -96,10 +96,6 @@
 
     public Config() {
     }
-    
-    static {
```

Comment:
It's bad idea, since URL object could be created during config file parsing.

Reply:
That should be fine as long as they're all created through `URIBuilder.create()`, which appears to be the case for all Redisson code. The patch will get applied there if any URL has an underscore in it. Am I missing something?

Reply:
URI created during configuration parsing though `Config.fromJSON` method, for example. Jackson parser creates URI object without `URIBuilder.create` method invocation.

Reply:
Ah, ok. Somehow that doesn't appear to be an issue though. I just tested `Config.fromJSON` without the patch and it's able to construct the URI object even with an underscore. It seems that Jackson is able to set the field without going through the normal check.

Reply:
> I just tested Config.fromJSON without the patch and it's able to construct the URI object even with an underscore

This is because of static initialization block in Config class.

Reply:
I tested with that block removed. I see now though, the problem isn't at construction time. It becomes an issue when `URI.getHost` is called, you get `null` back if there was an underscore. So this approach won't work.